cdap-docs/faq/source/cdap-startup.rst

```diff
@@ -0,0 +1,255 @@
+-----------------------------------------------
+If you've installed the ``cdap-cli`` RPM or DEB, it's located under ``/opt/cdap/cli/bin``.
+If you have installed CDAP manually (without using Cloudera Manager or Apache Ambari),
+you can add this location to your PATH to prevent the need for specifying the entire script every time.
+
+**Note:** These commands will list the contents of the package ``cdap-cli``, once it has
+been installed::
+
+  rpm -ql cdap-cli
+  dpkg -L cdap-cli
+
+.. _faq-installation-startup-memory-core-requirements:
+
+What are the memory and core requirements for CDAP?
+---------------------------------------------------
+The settings are governed by two sources: CDAP and YARN, and the requirements are
+:ref:`described here <install-hardware-memory-core-requirements>`.
+
+Can a current CDAP installation be upgraded more than one version?
+------------------------------------------------------------------
+In general, no. (The exception is an upgrade from 2.8.x to 3.0.x.)
+This table lists the upgrade paths available for different CDAP versions:
+
++---------+---------------------+
+| Version | Upgrade Directly To |
++=========+=====================+
+| 3.1.x   | 3.2.x               |
++---------+---------------------+
+| 3.0.x   | 3.1.x               |
++---------+---------------------+
+| 2.8.x   | 3.0.x               |
++---------+---------------------+
+| 2.6.3   | 2.8.2               |
++---------+---------------------+
+
+If you are doing a new installation, we recommend using the current version of CDAP.
+
+
+Are at least two machines really required for CDAP services?
+------------------------------------------------------------
+The CDAP components are independently scalable, so you can install from 1 to *N* of each
+component on any combination of nodes.  The primary reasons for using at least two
+machines are for HA (high availability) and for ``cdap-router``'s data ingest capacity.
+
+It is not necessary to install all components on both machines; you could install just the
+CDAP UI on a third machine with other components on the second node. You can install each
+component on a separate machine (or more) if you choose. The :ref:`HA [High Availability]
+Environment diagram <deployment-architectures-ha>` gives just one possible
+configuration.
+
+The HiveServer2 defaults to 10000; what should I do?
+----------------------------------------------------
+By default, CDAP uses port 10000. If port 10000 is being used by another service, simply
+change the ``router.bind.port`` in the ``cdap-site.xml`` to another available port. Since
+in the Hadoop ecosystem, HiveServer2 defaults to 10000, we are considering 
+`changing the router default port <https://issues.cask.co/browse/CDAP-1696>`__. 
+
+If you use Apache Ambari to install CDAP, it will detect this and run the CDAP Router on
+port 11015. Another solution is to simply run the CDAP Router on a different host than
+HiveServer2.
```

Comment:
If you use the Cloudera Manager CSD to install CDAP, the default ``router.bind.port`` is 11015.